Fixes part of #18442: Remove deprecated edit state content suggestion model

masterboy376 opened the pull request:
## Overview

1. This PR fixes part of #18442.
2. This PR does the following: Introduces a new beam job to delete deprecated edit state content suggestion from datastore.

## Essential Checklist

- [x] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [x] I have followed the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- [x] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I don't have permissions to assign reviewers directly).
- [x] The linter/Karma presubmit checks pass on my local machine, and my PR follows the [coding style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide)).
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)

## Testing doc (for PRs with Beam jobs that modify production server data)

- [x] A testing doc has been written: [Testing doc](https://docs.google.com/document/d/1GjH89H0qy8xToU57Ia5mEpmq31xorNRb3LameVNKtYg/edit?usp=sharing)
- [ ] _(To be confirmed by the server admin)_ All jobs in this PR have been verified on a live server, and the PR is safe to merge.

## Proof that changes are correct

![Screenshot (130)](https://github.com/oppia/oppia/assets/92575005/9cde6a59-e004-4df6-b401-8a858573fe0d)

## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

masterboy376 committed:
wrote and regidtered beam job

masterboy376 committed:
fic failing checks

masterboy376 committed:
fixed some linting issues

masterboy376 commented:
@seanlip @vojtechjelinek @DubeySandeep @kevintab95, PTAL

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

seanlip commented:
@masterboy376 I have not seen this testing doc before, but it is listed as me being the server admin and all the dropdowns in the reviewer checklist are marked as Yes. However I don't think this has actually been reviewed by any of us yet (I don't even have access to edit the doc).

Please read and follow the instructions in the release testing doc template carefully, and do not self-approve things and attribute those approvals to other people. That is not OK. (I am also not supposed to be the server admin for this.)

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

masterboy376 commented:
> @masterboy376 I have not seen this testing doc before, but it is listed as me being the server admin and all the dropdowns in the reviewer checklist are marked as Yes. However I don't think this has actually been reviewed by any of us yet (I don't even have access to edit the doc).
> 
> Please read and follow the instructions in the release testing doc template carefully, and do not self-approve things and attribute those approvals to other people. That is not OK. (I am also not supposed to be the server admin for this.)

Pardon @seanlip, I misunderstood the testing doc instruction before. I have fixed it.

masterboy376 commented:
@kevintab95 and @U8NWXD , just polite reminder PTAL.

DubeySandeep reviewed (approved):
LGTM for changes in the (2) codeowner files!

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

masterboy376 commented:
@kevintab95 friendly ping, PTAL

vojtechjelinek commented on core/storage/base_model/gae_models_test.py:
```diff
@@ -136,10 +136,10 @@ def test_generic_query_put_get_and_delete_operations(self) -> None:
         all_models = base_models.BaseModel.get_all()
         self.assertEqual(all_models.count(), 0)
-        self.assertEqual(model_id, 4)
+        self.assertEqual(model_id, 12)
```
Why is this change needed?

masterboy376 replied:
@vojtechjelinek, on my local machine the test worked fine with model_id=4 but during github ci checks this test fails with "AssertionError: 12 != 4". Could you please help me figure out what might be causing this?

seanlip replied:
@masterboy376 Please at least write a [debugging doc](https://github.com/oppia/oppia/wiki/Debugging-Docs) and outline what you've tried to investigate. Reviewers shouldn't be doing "debugging from scratch" for your PR.

masterboy376 replied:
Hello @seanlip and @vojtechjelinek ,
I fixed the bug by placing new test file in a different test shard. Here is the [debugging doc](https://docs.google.com/document/d/1FUN8h3INc2S3HxJERfKxzJF5HSV58-34guuz9TSakwQ/edit?usp=sharing) for the same.

vojtechjelinek reviewed (changes requested):
Thanks! Just one small nit?

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

masterboy376 committed:
address reviewer comments

masterboy376 committed:
modified test shards

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

masterboy376 commented:
@seanlip @kevintab95 @vojtechjelinek PTAL

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

masterboy376 commented:
@kevintab95 PTAL

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

kevintab95 commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs_test.py:
```diff
@@ -0,0 +1,284 @@
+    def setUp(self) -> None:
+        super().setUp()
+
+        self.suggestion_1_model: suggestion_models.GeneralSuggestionModel = (
```
Instead of numbering the variables, is it possible to make the names provide more context about what it represents? e.g. state content suggestions & question suggestion.

masterboy376 replied:
done

kevintab95 reviewed (changes requested):
Thanks @masterboy376! Just one nit from my end.

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

masterboy376 committed:
addressed reviewer comments.

masterboy376 committed:
addressed reviewer comments

masterboy376 commented:
@kevintab95, PTAL

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

seanlip commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs.py:
```diff
@@ -0,0 +1,123 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Deletion Jobs for edit state content suggestion models"""
```
Deletion jobs ...

End the docstring with a period.

masterboy376 replied:
done

seanlip commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs.py:
```diff
@@ -0,0 +1,123 @@
+                    lambda model: ((
+                        model.suggestion_type == (
+                            feconf.SUGGESTION_TYPE_EDIT_STATE_CONTENT)) and (
+                                model.target_type == (
```
Do you actually need this check?

masterboy376 replied:
Here, first we are fetching all the general suggestion models the filtering needed models using `model.suggestion_type == feconf.SUGGESTION_TYPE_EDIT_STATE_CONTENT`. However, here we can remove later filter condition that is `model.target_type == feconf.ENTITY_TYPE_EXPLORATION`, if all `SUGGESTION_TYPE_EDIT_STATE_CONTENT` models are of target type `ENTITY_TYPE_EXPLORATION`, and this seems true after the list of `SUGGESTION_TYPE_EDIT_STATE_CONTENT` models that you provided earlier.

seanlip commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs.py:
```diff
@@ -0,0 +1,123 @@
+        )
+
+        return (
+                suggestion_edit_state_content_model_to_delete,
```
Deindent this and the next line by 4

masterboy376 replied:
done

seanlip commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs.py:
```diff
@@ -0,0 +1,123 @@
+                GetDeprecatedSuggestionEditStateContentModels())
+        )
+
+        unused_models_deletion = (
```
unused_models_deletion_result

Though, if it's unused, then why bother to declare it in the first place?

masterboy376 replied:
We need to assign expression `"suggestion_edit_state_content_model_to_delete | 'Extract keys' >> beam.Map(lambda model: model.key) | 'Delete models' >> ndb_io.DeleteModels()"` to something to avoid linting error reading `"expression is assigned to nothing"`, so, I assigned it to `unused_models_deletion`.

seanlip replied:
OK, can you rename to unused_models_deletion_result then (as stated in the earlier comment)?

seanlip commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs.py:
```diff
@@ -0,0 +1,123 @@
+        )
+
+        return (
+            (
```
I don't think you need this parens and the indentation below?

masterboy376 replied:
done

seanlip commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs.py:
```diff
@@ -0,0 +1,123 @@
+            (
+                suggestion_edit_state_content_model_to_delete_result,
+            )
+            | 'Merge results' >> beam.Flatten()
```
Just wondering, why do we not report these the same way as in the audit job?

masterboy376 replied:
Totally agree, we actually don't need to use beam.Flatten() here.
Made required changes.

seanlip commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs_test.py:
```diff
@@ -0,0 +1,286 @@
+            final_reviewer_id: str,
+            language_code: None | str
+            ) -> suggestion_model:
+        """creates new suggestion.
```
creates --> Creates

masterboy376 replied:
done

seanlip commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs_test.py:
```diff
@@ -0,0 +1,286 @@
+            final_reviewer_id: str,
+            language_code: None | str
+            ) -> suggestion_model:
+        """creates new suggestion.
```
Ditto.

masterboy376 replied:
done

seanlip commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs_test.py:
```diff
@@ -0,0 +1,286 @@
+        Returns:
+            GeneralSuggestionModel. Returns newly created
+            GeneralSuggestionModel.
+            """
```
Deindent by 4

masterboy376 replied:
done

seanlip commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs_test.py:
```diff
@@ -0,0 +1,286 @@
+
+        self.assertEqual(
+            len(suggestion_model.query_suggestions(
+                queries)), 2)
```
You should be able to move this to the end of the previous line?

masterboy376 replied:
done

masterboy376 committed:
addressed reviewers comment

vojtechjelinek reviewed (approved):
LGTM!

masterboy376 commented:
Hello @seanlip, addressed your review comments PTAL.

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

seanlip commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs.py:
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 
-"""Deletion Jobs for edit state content suggestion models"""
+"""Deletion Jobs for edit state content suggestion models."""
```
See my previous comment: "Jobs" should start with a lowercase letter.

masterboy376 replied:
Jobs -> jobs

Done

seanlip reviewed (commented):
Thanks! Looks fine, just minor notes. 

Also, in future, please double-check to ensure that previous comments are properly addressed; all remaining comments are repeats of previous comments that weren't fully addressed.

masterboy376 committed:
addressed reviewer comments

masterboy376 commented:
> Thanks! Looks fine, just minor notes.
> 
> Also, in future, please double-check to ensure that previous comments are properly addressed; all remaining comments are repeats of previous comments that weren't fully addressed.

Thanks @seanlip, I apologize for missing out some comment. I have now addressed all the remaining reviews. PTAL

seanlip reviewed (approved):
Thanks -- LGTM from my end.

kevintab95 commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs_test.py:
```diff
@@ -0,0 +1,280 @@
+    def setUp(self) -> None:
+        super().setUp()
+
+        self.edit_state_content_suggestion_model_1: suggestion_model = (
```
```suggestion
        self.state_content_suggestion_model_1: suggestion_model = (
```
Ditto below

masterboy376 replied:
done

kevintab95 commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs_test.py:
```diff
@@ -0,0 +1,280 @@
+                None
+            ))
+
+        self.create_translation_suggestion_model: suggestion_model = (
```
```suggestion
        self.translation_suggestion_model: suggestion_model = (
```

masterboy376 replied:
done

kevintab95 commented on core/jobs/batch_jobs/suggestion_edit_state_content_deletion_jobs_test.py:
```diff
@@ -0,0 +1,280 @@
+                'hi'
+            ))
+
+        self.add_question_suggestion_model: suggestion_model = (
```
```suggestion
        self.question_suggestion_model: suggestion_model = (
```

In general, avoid starting variable names with verbs.

masterboy376 replied:
done

masterboy376 committed:
addressed reviewer comments

kevintab95 reviewed (approved):
LGTM, thanks @masterboy376!

masterboy376 committed:
Merge remote-tracking branch 'upstream/develop' into remove_deprecated_edit_state_content_suggestion_model

chris7716 reviewed (changes requested):
Assigning my self as the server admin to run this job

chris7716 reviewed (approved):
Ran it successfully on backup server.
